### ISSUE TITLE
[Enhancement]: Default to the currency setting when currency changed

### ISSUE THREAD

**Issue by kidinov** [woothemes/woocommerce#40016]
### Describe the solution you'd like

When switching currencies, settings remain unchanged, which leads in many cases to the wrong and confusing for both merchant's and buyers' money display. Defaults should be applied, including "Number of decimals."

Also, to make it easy to use and less confusing, we may want to hide all the settings except the currency itself in some kind of "advanced" hidden section of the screen.

Validation of the input for this field won't hurt as well

<img width="686" alt="image" src="https://github.com/woocommerce/woocommerce/assets/4923871/e82ab5bd-6e56-4431-90a3-006978403788">

<img width="300" alt="image" src="https://github.com/woocommerce/woocommerce/assets/4923871/6693cf5e-68a0-4176-b697-f61e41d5f373">

<img width="343" alt="image" src="https://github.com/woocommerce/woocommerce/assets/4923871/8e63d541-79a0-4bc6-b776-fdf02bdc3a75">

### Additional context

Slack convo - p1689340025255369-slack-C0E1AV8T0

**Comment by pierorocca** [comment]
@vbelolapotkov did you mean to assign this to me? This looks like a merchant mccy setting?

**Comment by vbelolapotkov** [comment]
@pierorocca yes, I did, probably a bit too early. Also pinging @bborman22 since his team now is the best fit here.

**Comment by pierorocca** [comment]
@vbelolapotkov I have the core currency functionality as a shared item between ME and BE per Dwain's post.

> Merchant/Shopper: WooCommerce Core Currency functionality. 

This enhancement is predominantly about the store setup improving the core UI usability for the merchant. I'd strongly prefer this one stay with the ME team.

**Comment by flakas** [comment]
Thanks for the suggestion!

I think this would benefit from the design team's perspective.

The currency controls decimals and currency position, so changing the currency may need to adjust those.
Separators are controlled by the store's locale, so I'm not sure it'd be right to change those on currency change.

Core already changes the number of decimals setting when a currency is chosen in onboarding, but we don't have the same in Settings.

So I'd appreciate design team's perspective on the solution shape. I can think of:
* Silently change position/decimals when currency changes
* If the merchant customized settings - prompt the merchant to adjust on currency change, showing the proposed value
* Something else?

Note that merchants may customize the values, so we'd need to account for them.
